calcom/cal.com
# fix: location for `@calcom/embed-react` types

## What does this PR do?



Fixes #9089

I think when we [upgraded Vite](https://github.com/calcom/cal.com/pull/7197), the location for generated types got changed somehow. This PR uses the correct location now.

Also, `floatingButton` config support is live but it's typings won't work if people are using getCalApi. It also fixes that by upgrading @calcom/embed-core


## Type of change


- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [Make a simple nextjs project.](https://nextjs.org/docs/pages/api-reference/create-next-app) in TypeScript
- Add @calcom/embed-react as a dependency
- Use [this sample code](https://codesandbox.io/s/floating-button-popup-react-8pwvk4?file=/src/App.tsx:140-554)
- Notice that you would recieve TypeScript errors
- Build packages/embeds/embed-react and copy the content to the nextjs project's node_modules/@calcom/embed-react. It should fix the issue. You can also simply try tweaking the types property of package.json in node_modules/@calcom/embed-react
<img width="605" alt="Screenshot 2023-05-24 at 6 14 20 PM" src="https://github.com/calcom/cal.com/assets/1780212/84f790c6-1703-483e-840c-5359ec91c7a4">

## Followup
- [ ]  This bug never came up during local testing as in local testing we are testing the source code and not the built code that's published. Created a ticket to be able to test that in CI https://github.com/calcom/cal.com/issues/9091

## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
